Homeworks/Homework_3/script.py

```diff
@@ -0,0 +1,369 @@
+    users_registered_age = [int(user['registered.age']) for user in rearranged_data[folder_decade][folder_country]]
+    users_id_name = [user['id.name'] for user in rearranged_data[folder_decade][folder_country]]
+
+    return f'{max(users_age)}_' \
+           f'{sum(users_registered_age) / len(users_registered_age)}_' \
+           f'{Counter(users_id_name).most_common(1)[0][0]}'
+
+
+def create_sub_folders(rearranged_data):
+    """
+    Create folders with csv files based on the new data format
+
+    :param rearranged_data: dict, data on the structure of which
+                            folders are created and csv files into folders are filled
+    """
+    for decade in rearranged_data:
+        decade_path = os.path.join(os.getcwd(), decade)
+        os.makedirs(decade_path)
+        for country in rearranged_data[decade]:
+            country_path = os.path.join(decade_path, country)
+            os.makedirs(country_path)
+
+            csv_file_name = create_file_name(rearranged_data, decade, country)
+
+            csv_file_path = os.path.join(country_path, csv_file_name)
+
+            write_new_data_to_csv(csv_path=csv_file_path + '.csv', csv_data=rearranged_data[decade][country])
+
+
+def remove_data_before_60_th():
+    """
+    In destination folder delete folders with users which were born before 1960-th
+    """
+    for dec in os.listdir(os.getcwd()):
+        if dec[0] < '6' and dec[0] != '0':
+            shutil.rmtree(os.path.join(os.getcwd(), dec))
+
+
+def log_folder_structure(destination_folder):
+    """
+    Log the data structure that is in destination folder
+
+    :param destination_folder: str, path to destination folder
+    """
+    for root, dirs, files in os.walk(destination_folder):
+        level = root.replace(destination_folder, '').count(os.sep)
+        indent = ' ' * 4 * level
+        logging.info(f'{indent}{os.path.basename(root)}/')
+        sub_indent = ' ' * 4 * (level + 1)
+        for f in files:
+            logging.info(f'{sub_indent}{f} (file)')
+        for d in dirs:
+            logging.info(f'{sub_indent}{d} (folder)')
+
+
+def archive_destination_folder(destination_folder_name):
+    """
+    Archive destination folder into zip archive
+
+    :param destination_folder_name: str, name of destination folder
+    """
+    shutil.make_archive(destination_folder_name, 'zip')
+
+
+def parse_command_line():
+    parser = argp.ArgumentParser()
+
+    parser.add_argument('--destination_folder', help='path to the folder where the output file will be placed',
+                        required=True)
+    parser.add_argument('filename', default='output', help='name of the output file')
+
+    group = parser.add_mutually_exclusive_group()
+
+    group.add_argument('--gender', default=None, help='filter the data by gender')
+    group.add_argument('--number_of_rows', default=None, type=int, help='filter the data by the number of rows')
```

Comment:
I thought if you didn't set required and skipped the key it should be None without explicit setting of default=None